### PR TITLE
add relative timestamp function

### PR DESCRIPTION
**Pull request by j-ogas**
Addresses #42.

Note, I alphabetized the import statements (unrelated to the issue at hand but it bothered me)

### CODE REVIEW

File: lib/pavilion/utils.py [hunk 1]
```diff
@@ -1,16 +1,38 @@
...
+import shutil
 import subprocess
 import sys
 import textwrap
-import shutil
-import itertools
 from collections import defaultdict
+from pathlib import Path
+
+def get_relative_timestamp(obj):
```
Comment (Paul-Ferrell) [inline lib/pavilion/utils.py]: `base_dt`

**Comment by j-ogas** [comment]
@pflarr Let me know when I can resume working on this. Last we chatted, we needed pav to pass an object (instead of string) for the transformation in `draw_tables` to work correctly.

File: lib/pavilion/utils.py [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
 import sys
 import textwrap
-import shutil
-import itertools
 from collections import defaultdict
+from pathlib import Path
+
+def get_relative_timestamp(base_dt):
+    """Print formatted time string based on the delta of time objects.
+    :param obj datetime object
```
Comment (Paul-Ferrell) [inline lib/pavilion/utils.py]: :param <type> <name>: <description>

:param datetime base_dt: The timestamp to format.

File: lib/pavilion/utils.py [hunk 3]
```diff
@@ -1,16 +1,39 @@
...
+    :rtype string
```
Comment (Paul-Ferrell) [inline lib/pavilion/utils.py]: :rtype: str

File: lib/pavilion/utils.py [hunk 4]
```diff
@@ -1,16 +1,39 @@
...
+    """
+    now = datetime.datetime.now()
+    format_ = ['%Y', '%b', '%a', '%H:%M:%S'] # year, month, day, time
+
```
Comment (Paul-Ferrell) [inline lib/pavilion/utils.py]: for i in range(len(format_)):
  if now.strftime(format_[i]) != ....:
     return now.strftime(" ".join(format_[i:])

File: lib/pavilion/utils.py [hunk 5]
```diff
@@ -222,8 +238,8 @@ class PavEncoder(json.JSONEncoder):
     def default(self, o):  # pylint: disable=E0202
         if isinstance(o, Path):
             return super().default(str(o))
```
Comment (Paul-Ferrell) [inline lib/pavilion/utils.py]: This is a bug (in my code, not yours). It should just `return str(o)` here.

File: lib/pavilion/utils.py [hunk 6]
```diff
@@ -222,8 +238,8 @@ class PavEncoder(json.JSONEncoder):
...
-        else:
-            return super().default(str(o))
+        if isinstance(o, (datetime.datetime)):
+            return o.isoformat()
```
Comment (Paul-Ferrell) [inline lib/pavilion/utils.py]: and `return super().default(o)` otherwise.

**Pull request merged**